src/identity/processor.rs

```diff
@@ -143,13 +143,18 @@ impl IdentityProcessor for OnChainIdentityProcessor {
             // Note that we don't have a way of queuing a root here for
             // finalization. so it's going to stay as "processed"
             // until the next root is mined. self.database.
-            self.database.mark_root_as_processed_tx(&root_hash).await?;
-            self.database.delete_batches_after_root(&root_hash).await?;
+            self.database
+                .mark_root_as_processed_and_delete_batches_tx(&root_hash)
+                .await?;
         } else {
             // Db is either empty or we're restarting with a new contract/chain
             // so we should mark everything as pending
-            self.database.mark_all_as_pending().await?;
-            self.database.delete_all_batches().await?;
+            retry_tx!(self.database.pool, tx, {
```

Comment:
It would be nice to put this into transactions.rs file in function to keep it aligned with second if branch. :)